src/utils/daos_hdlr.c

```diff
@@ -1613,19 +1613,19 @@ parse_filename_dfs(const char *path, char **_obj_name, char **_cont_name)
 		return -EINVAL;
 
 	if (strcmp(path, "/") == 0) {
-		*_cont_name = strdup("/");
+		D_STRNDUP(*_cont_name, "/", 2);
 		if (*_cont_name == NULL)
 			return -ENOMEM;
 		*_obj_name = NULL;
 		return 0;
 	}
-	f1 = strdup(path);
+	D_STRNDUP(f1, path, strlen(path) + 1);
 	if (f1 == NULL) {
 		rc = -ENOMEM;
 		goto out;
 	}
 
-	f2 = strdup(path);
+	D_STRNDUP(f2, path, strlen(path) + 1);
```

Comment:
@dsikich Maybe calculate `strlen(path)` once and use it in both places?
For what it's worth, this function will be refactored when the DFS wrappers are created as part of [DAOS-6098](https://jira.hpdd.intel.com/browse/DAOS-6098)